### PR TITLE
Feature/crocus noah3.9

### LINKED ISSUE #1025: Add support for coupling Crocus and Noah39
We have already developed an infrastructure to couple the sub LSM with different LSMs. As a pilot study, we coupled Crocus and Noah-MP401. Here we add support for coupling Crocus and Noah39 LSM.  

@LIS-navari

### PR DESCRIPTION
**Pull request by LIS-navari**
### Description
We have already developed an infrastructure to couple the sub LSM with different LSMs. As a pilot study, we coupled Crocus and Noah-MP401. Here we add support for coupling Crocus and Noah39 LSM.  

Resolves #1025 

Replace this text with a concise description of *what* was changed and *why*.

### LDT 
/discover/nobackup/mnavari/my_projects/thp/submodel/noah39/procLSM
ldt.config.hymap

### LIS
/discover/nobackup/mnavari/my_projects/thp/submodel/noah39/nldas
lis.config
lis.job

### CODE REVIEW

**Comment by bmcandr** [comment]
Hi @LIS-navari, can you please open permissions on the `output/` directory in your LIS testcase? Thanks!

**Comment by LIS-navari** [comment]
Hi @bmcandr, You should have access to the output. Thanks

File: lis/plugins/LIS_lsmrouting_pluginMod.F90 [hunk 1]
```diff
@@ -160,6 +162,9 @@ subroutine LIS_lsmrouting_plugin
    call registerlsmroutinggetrunoff(trim(LIS_noah39Id)//"+"//&
         trim(LIS_HYMAProuterId)//char(0), &
         noah39_getrunoffs_mm)
+!   call registerlsmroutinggetsws(trim(LIS_noah39Id)//"+"//&
```
Comment (bmcandr) [inline lis/plugins/LIS_lsmrouting_pluginMod.F90:165]: Can we delete this entry? Looks like it was moved down to lines 261.

File: lis/surfacemodels/land/noah.3.9/routing/noah39_getrunoffs_hymap2.F90 [hunk 2]
```diff
@@ -0,0 +1,96 @@
+!-----------------------BEGIN NOTICE -- DO NOT EDIT-----------------------
+! NASA Goddard Space Flight Center Land Information System (LIS) v7.2
+!
+! Copyright (c) 2015 United States Government as represented by the
+! Administrator of the National Aeronautics and Space Administration.
+! All Rights Reserved.
+!-------------------------END NOTICE -- DO NOT EDIT-----------------------
```
Comment (bmcandr) [inline lis/surfacemodels/land/noah.3.9/routing/noah39_getrunoffs_hymap2.F90:7]: Update copyright notice

File: lis/surfacemodels/land/noah.3.9/routing/noah39_getsws_hymap2.F90 [hunk 3]
```diff
@@ -0,0 +1,49 @@
+!-----------------------BEGIN NOTICE -- DO NOT EDIT-----------------------
```
Comment (bmcandr) [inline lis/surfacemodels/land/noah.3.9/routing/noah39_getsws_hymap2.F90:1]: Update copyright notice

**Review by bmcandr — CHANGES_REQUESTED** [review]
Hi @LIS-navari, sorry for the delay. I have generated outputs with the latest Intel 2021 environment and see minor differences that I'm assuming are due to using different compiler versions (I see a LIS_19 in your testcase directory). I'm re-running the testcase with the Intel 19 environment to confirm.

On the code review front, things look good overall. See my one question ~above~ below about deleting some commented out lines. I'll handle the copyright notices.

Two questions:
* Is Crocus ready to be enabled in make/default.cfg? In #697 we decided that more work was needed so I just wanted to circle back on that.
* (if "yes" to the above question) Should Crocus be added to the documentation anywhere?

**Comment by emkemp** [comment]
Hi Mahdi:

I dusted off this pull request and performed some testing.  I get identical results in the SURFACEMODEL history file, but I see some differences in the ROUTING file (both valid 00Z 2 Jan 2021).  I'm assuming these differences are due to my use of the Intel 2021 compiler.

I updated the copyright blocks in the new source code files you added to match what is in the master branch (LISf 7.4, copyright 2022).

The documentation still needs to be updated to reflect the CROCUS entries, as well as the subLSM model entry.  I need you to provide these updates before this pull request is "ready" to be merged.

**Comment by emkemp** [comment]
*PING*

Hi @mahdi:

This pull request still needs additional documentation before we can accept your changes into master.

**Comment by LIS-navari** [comment]
Hi @emkemp 

I have completed the documentation.

Thanks
Mahdi

**Comment by emkemp** [comment]
Thanks @LIS-navari.

I need to coordinate with @jvgeiger on how to accept your test case before accepting the pull request.

**Comment by emkemp** [comment]
Test cases copied to /discover/nobackup/projects/lis/PR_TESTCASES/1026.  I approve.

**Pull request merged**